### PR TITLE
final_aks_config

### PR DESCRIPTION
**Pull request by wtomaz808**
## Issue number

#000

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [ ] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,140 @@
-# Azure Template Overlay
+# Terraform-Azurerm-Overlays-Kubernetes
 
 [![Changelog](https://img.shields.io/badge/changelog-release-green.svg)](CHANGELOG.md) [![Notice](https://img.shields.io/badge/notice-copyright-yellow.svg)](NOTICE) [![MIT License](https://img.shields.io/badge/license-MIT-orange.svg)](LICENSE) [![TF Registry](https://img.shields.io/badge/terraform-registry-blue.svg)](https://registry.terraform.io/modules/azurenoops/overlays-template/azurerm/)
 
-This Overlay terraform module can create a Template and manage related parameters to be used in a [SCCA compliant Network](https://registry.terraform.io/modules/azurenoops/overlays-hubspoke/azurerm/latest).
+## Overview
 
+This overlay module deploys a Azure Kubernetes Service - Private Cluster - Kubenet suitable for hosting docker containers apps. The cluster will be deployed to the Hub/Spoke shared services resource group or Tier 3 spoke using default naming unless alternative values are provided at run time.
```
Comment (jftl6y) [inline README.md]: containers ->container

File: README.md [hunk 2]
```diff
@@ -1,14 +1,140 @@
...
+
+Read on to understand what this example does, and when you're ready, collect all of the pre-requisites, then deploy the example.
 ## SCCA Compliance
 
-This module can be SCCA compliant and can be used in a SCCA compliant Network. Enable private endpoints and SCCA compliant network rules to make it SCCA compliant.
+This overlay module will deploy a SCCA compliant, Private AKS Cluster. This is accomplished by isolating the the control plane from the user plane. Egress is controlled by creating a UserDefinedRoute that direct all tracffic out of the Hub Firewall. Ingress in controlled by directing all inbound traffic through a Azure Loadbalancer. The Private AKS Cluster is also configured with private endpoints along with SCCA compliant network rules.
```
Comment (jftl6y) [inline README.md]: direct->directs

"through an Azure Load Balancer"

File: README.md [hunk 3]
```diff
@@ -1,14 +1,140 @@
...
+
+Ref: Azure NoOps Hub Module [SCCA compliant Network](https://registry.terraform.io/modules/azurenoops/overlays-hubspoke/azurerm/latest).
+For more information, please read the [SCCA documentation](). 
+
+## Deploy Azure Kubernetes Service - Private Cluster - Kubenet
+
+The docs on Azure Kubernetes Service (AKS): <https://docs.microsoft.com/en-us/azure/aks/>.  By default, this overlay will deploy resources into standard default hub/spoke subscriptions and resource groups.
```
Comment (jftl6y) [inline README.md]: Replace link with learn equivalent.  "The docs" is informal, rephrase

File: README.md [hunk 4]
```diff
@@ -1,14 +1,140 @@
...
+This overlay module will deploy a SCCA compliant, Private AKS Cluster. This is accomplished by isolating the the control plane from the user plane. Egress is controlled by creating a UserDefinedRoute that direct all tracffic out of the Hub Firewall. Ingress in controlled by directing all inbound traffic through a Azure Loadbalancer. The Private AKS Cluster is also configured with private endpoints along with SCCA compliant network rules.
+
+Ref: Azure NoOps Hub Module [SCCA compliant Network](https://registry.terraform.io/modules/azurenoops/overlays-hubspoke/azurerm/latest).
+For more information, please read the [SCCA documentation](). 
+
+## Deploy Azure Kubernetes Service - Private Cluster - Kubenet
+
+The docs on Azure Kubernetes Service (AKS): <https://docs.microsoft.com/en-us/azure/aks/>.  By default, this overlay will deploy resources into standard default hub/spoke subscriptions and resource groups.  
+
+The Docs on AKS Custome egress: <https://learn.microsoft.com/en-us/azure/aks/egress-udr/>
```
Comment (jftl6y) [inline README.md]: Change "The Docs" to something less informal

Custome->Custom

File: README.md [hunk 5]
```diff
@@ -1,14 +1,140 @@
...
+
+The subscription and resource group can be changed by providing the resource group name (Param: parTargetSubscriptionId/parTargetResourceGroup) and ensuring that the Azure context is set the proper subscription.
```
Comment (jftl6y) [inline README.md]: "set to the proper subscription"

File: README.md [hunk 6]
```diff
@@ -1,14 +1,140 @@
...
+ ![Private Azure Kubernetes Service](./media/aksprivatecluster_vnetpeering.jpg)
+## Fully Private AKS Cluster Using Private Endpoint
+ ![Private Azure Kubernetes Service](./media/AKSPrivateClusterTier3.jpg)
+## Outbound Type LoadBalancer
+ ![Private Azure Kubernetes Service](./media/aksprivatecluster_lb.jpg)
+ 
+
+## About Azure Container Registry
+
+The docs on Azure Container Registry: <https://docs.microsoft.com/en-us/azure/container-registry/>. By default, this workload uses the Azure Container Registry workload to deploy resources into the AKS workload spoke.
```
Comment (jftl6y) [inline README.md]: Replace informal "The docs" with Microsoft Learn
Replace URL with learn version https://learn.microsoft.com/en-us/azure/container-registry/

File: README.md [hunk 7]
```diff
@@ -1,14 +1,140 @@
...
+ ![Private Azure Kubernetes Service](./media/aksprivatecluster_lb.jpg)
+ 
+
+## About Azure Container Registry
+
+The docs on Azure Container Registry: <https://docs.microsoft.com/en-us/azure/container-registry/>. By default, this workload uses the Azure Container Registry workload to deploy resources into the AKS workload spoke. 
+
+## About Azure Kubernetes Service - Private Cluster
+
+The docs on Azure Kubernetes Service: <https://docs.microsoft.com/en-us/azure/aks/>.  this workload uses the Azure Kubernetes Service - Cluster workload to deploy resources into the AKS workload spoke.
```
Comment (jftl6y) [inline README.md]: Replace informal "The docs" with Microsoft Learn
Replace URL with learn version https://learn.microsoft.com/en-us/azure/aks/

File: resources.kubernetes.identity.tf [hunk 8]
```diff
@@ -39,7 +39,7 @@ resource "azurerm_role_assignment" "route_table_network_contributor" {
 resource "azurerm_role_assignment" "aks_identity_contributor" {
   principal_id          = azurerm_user_assigned_identity.aks[0].principal_id #azurerm_user_assigned_identity.aks_identity.principal_id
   role_definition_name  = "Contributor"  # You can use a more specific role if needed
-  scope                 = "/subscriptions/df79eff1-4ca3-4d21-9c6b-64dd15c253e8/resourceGroups/tf-anoa-gov-rg-aks/providers/Microsoft.Network/routeTables/rt-aks_egress" #azurerm_route_table.rt.id
+  scope                 = data.azurerm_route_table.rt.id #"/subscriptions/df79eff1-4ca3-4d21-9c6b-64dd15c253e8/resourceGroups/tf-anoa-gov-rg-aks/providers/Microsoft.Network/routeTables/rt-aks_egress"
```
Comment (jftl6y) [inline resources.kubernetes.identity.tf]: It looks like here we have a subscription ID (commented but still visible)

**Review by jftl6y — COMMENTED** [review]
See pending comments

File: resources.kubernetes.identity.tf (hunk 8 above)
Comment (wtomaz808) [reply resources.kubernetes.identity.tf]: ah crud that comment should be removed..

File: README.md (hunk 7 above)
Comment (wtomaz808) [reply README.md]: concur

File: README.md (hunk 6 above)
Comment (wtomaz808) [reply README.md]: concur

File: README.md (hunk 5 above)
Comment (wtomaz808) [reply README.md]: concur yes it appears we left out"to" "set to the proper subscription"

File: README.md (hunk 4 above)
Comment (wtomaz808) [reply README.md]: maybe just "Documentation: " and yes custom is misspelled..

File: README.md (hunk 3 above)
Comment (wtomaz808) [reply README.md]: maybe we just use "Documentation: "

File: README.md (hunk 2 above)
Comment (wtomaz808) [reply README.md]: misspelling concur

File: README.md (hunk 1 above)
Comment (wtomaz808) [reply README.md]: concur misspelling

**Review by wtomaz808 — COMMENTED** [review]
I made the changes you suggested..

**Review by jftl6y — APPROVED** [review]

**Pull request merged**